src/Extension/Extension.php

```diff
@@ -203,16 +211,30 @@ public function setVersion($version)
      * @param array $extensionSet: An associative array where keys are the composer package names
      *                             of installed extensions. Used to figure out which dependencies
      *                             are flarum extensions.
+     * @param array $enabledIds:   An associative array where keys are the composer package names
+     *                             of enabled extensions. Used to figure out optional dependencies.
      */
-    public function calculateDependencies($extensionSet)
+    public function calculateDependencies($extensionSet, $enabledIds)
     {
         $this->extensionDependencyIds = (new Collection(Arr::get($this->composerJson, 'require', [])))
             ->keys()
+            ->merge(Arr::get($this->composerJson, 'extra.flarum-extension.optional-dependencies', []))
             ->filter(function ($key) use ($extensionSet) {
                 return array_key_exists($key, $extensionSet);
-            })->map(function ($key) {
+            })
+            ->map(function ($key) {
+                return static::nameToId($key);
+            })
+            ->toArray();
+
+        $this->optionalDependencyIds = (new Collection(Arr::get($this->composerJson, 'extra.flarum-extension.optional-dependencies', [])))
+            ->map(function ($key) {
                 return static::nameToId($key);
```

Comment:
Should optional dependencies be provided as composer package names or Flarum extension IDs? I'm leaning towards switching it to the latter since this is Flarum-specific config

Reply:
Hm... multiple extensions can technically resolve to the same ID, so I feel like we'd want dependencies to use the package name. Specially as that's used for required dependencies (ignoring the fact that uses composer require in composer.json)

Reply:
Thing is, the composer package name isn't actually used anywhere in calculations; in the lines directly below here, it turns the composer package names into flarum extension IDs anyway.

Although on the other hand, if we ever decide to switch back to using pure composer package names, having that be the provided format now will reduce breaks then.

Reply:
I would lean towards using package names, just so that we're using one single format when referring to extensions in composer.json (consistency).

Reply:
In that case, let's keep as is